### PR TITLE
Ayomiku cu #860r58q5h enable changing filters on the food detail page

### PR DESCRIPTION
**Pull request by Ayomikuolatunji**

### CODE REVIEW

File: src/helpers/utils.ts [hunk 1]
```diff
@@ -11,3 +15,143 @@ export const spaceToDashes = (name: string) => {
     .replace(/[^\w]+/g, "-");
   return pathname.endsWith("-") ? pathname.substr(0, pathname.length - 1) : pathname;
 };
+
+export const getValueRounded = (amount: number) => {
+  return Math.round(amount * 100) / 100;
+};
+
+export function getAgeRangesForLifeStage(selectedLifeStageName: string = "Females"): RdiAge[] {
```
Comment (hhimanshu) [inline src/helpers/utils.ts:23]: lifeStage can be a type rather than string

File: src/helpers/utils.ts [hunk 2]
```diff
@@ -11,3 +15,143 @@ export const spaceToDashes = (name: string) => {
...
+          ? ageRange
+          : null;
+
+      default:
+        return ageRange;
+    }
+  });
+  return ageRanges.filter((range) => range !== null) as RdiAge[];
+}
+export function setSelectedAgeByGender(
```
Comment (hhimanshu) [inline src/helpers/utils.ts]: shall we call this method `setDefaultSelectedAgeForGender`

File: src/components/FoodNutritionCard/FoodNutritionCard.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ const FoodNutritionCard = () => {
                                 </p>
                               </div>
                               <p className="prose-bodyMedium text-dark font-400">
-                                {row.rdi?.amount ? `${row.rdi.unit}%` : ""}
+                                {row.rdi?.amount ? `${row.rdi.amount}%` : ""}
```
Comment (hhimanshu) [inline src/components/FoodNutritionCard/FoodNutritionCard.tsx]: For %, why are we looking for `amount` and not `percentDaily`?

File: src/helpers/utils.ts (hunk 1 above)
Comment (Ayomikuolatunji) [reply src/helpers/utils.ts:23]: fixed

Comment (hhimanshu) [reply src/helpers/utils.ts:23]: I still see the type as string. Why don't we have a something like this?
```ts
type LifeStage = "infants" | "children" | "males" | "females" | "pregnancy" | "lactation"
```
and use this type as argument here?

File: src/helpers/utils.ts (hunk 2 above)
Comment (Ayomikuolatunji) [reply src/helpers/utils.ts]: done

File: src/helpers/utils.ts (hunk 1 above)
Comment (Ayomikuolatunji) [reply src/helpers/utils.ts:23]: Done

Comment (Ayomikuolatunji) [reply src/helpers/utils.ts:23]: Done

File: src/components/FoodDetailsSummary/FoodDetailsSummary.tsx [hunk 4]
```diff
@@ -1,45 +1,64 @@
...
+                item.nutrient.name === "Protein" ||
+                item.nutrient.name === "Sodium"
+            )
+            .map((item, idx) => (
+              <div key={idx} className="mr-[35px]">
+                <div className="mt-2 flex justify-center items-center flex-col">
+                  <div className="w-[50px] h-[50px]">
+                    <CircularProgressbarWithChildren
+                      strokeWidth={10}
+                      value={item.rdi?.amount ? Math.ceil(item.percentDaily as number) : 0}
```
Comment (hhimanshu) [inline src/components/FoodDetailsSummary/FoodDetailsSummary.tsx]: You calculation for percentDaily is different at different place. I would say create a function called `getPercentDaily(amount: number)` and use that everywhere

**Review by hhimanshu — APPROVED** [review]

**Pull request merged**